src/App/DocumentObject.cpp

```diff
@@ -282,6 +282,16 @@ std::string DocumentObject::getFullName() const {
     return name;
 }
 

```

Comment:
> If the real name of the document is in foreign languages, such as JP, it is not correctly coded (document name appears as underscores).

It has nothing to do with incorrect encoding but the fact that for Python identifiers only a small number of characters were allowed. That's why all other characters are replaced with underscores. Now with Py3 any unicode characters are allowed and there is already a request to lift the current restrictions.